packages/collector/Collector_Content.php

```diff
@@ -0,0 +1,46 @@
+<?php
+function collector_iterate_directory($path, $prefix, $callback)
+{
+	$handle = opendir($path);
+
+	while($entry = readdir($handle))
+	{
+		if($entry === '.' || $entry === '..' || substr($entry, 0, 4) === '.git')
+		{
+			continue;
+		}
+
+		$realPath = realpath($path . '/' . $entry);
+		$packPath = substr($realPath, strlen($prefix));
+
+		$callback($realPath, $packPath);
+	}
+}
+
+function collector_export_tree()
+{
+	$flat = [];
+
+	$callback = function ($realPath, $packPath) use (&$flat, &$callback) {
+		if(is_file($realPath))
+		{
+			$flat[$packPath] = (object)[
+				'realPath' => $realPath,
+				'mode' => sprintf('%o', fileperms($realPath)),
```

Comment:
nice! I'm not sure if Emscripten supports that, but it should still be handy in `wp-now` eventually